analyzer/src/main/kotlin/managers/Pub.kt

```diff
@@ -114,6 +115,9 @@ class Pub(
         ) = Pub(type, analysisRoot, analyzerConfig, repoConfig)
     }
 
+    // TODO: Maybe enable Gradle functionality only if the plugin is present, and warn if it is not.
```

Comment:
> What's the reason to not just let the Pub plugin depend on the Gradle plugin?

It just seemed unnecessary to do to me.

> do you think there is a valid third-party use case where people want to use the Pub plugin but don't want to have the Gradle plugin on the classpath?

I don't know. I agree it's probably unlikely, but architecture-wise it seems easy to not depend on the Gradle-class hard-coded, so it felt right to not do it.

Reply:
There is no hard requirement for Pub to have Gradle available.
This is only the case for (most?) Flutter projects, other than that, Pub is a package manager that can and is used independently.

I think it makes sense to not have a hard Gradle dependency.

Reply:
> I think it makes sense to not have a hard Gradle dependency.

I've already implemented that in the most recent PR iteration.

Reply:
> Also, how would you actually implememnt that? Even if the Pub-PM Gradle project would depend on the Gradle-PM Gradle project, no one forces users of ORT to actually put the Gradle-PM JAR in the classpath when running ORT. So I believe we should have that case gracefully.

I'm not sure I can follow. If it's a dependency, build tools like Gradle or Maven would add that dependency to the classpath automatically so users would have to actively exclude the dependency to not have it on the classpath. But if you exclude a transitive dependency you should not be surprised if it causes issues.

> I don't know. I agree it's probably unlikely, but architecture-wise it seems easy to not depend on the Gradle-class hard-coded, so it felt right to not do it.

I don't disagree with that, but it seemed like unnecessary work to me to support an esoteric use case. However, it's done now already. But this also means that the upcoming Pub plugin should now test both scenarios, one with Gradle on the classpath and one without, because otherwise we can't claim that both are supported.

Reply:
> I'm not sure I can follow.

Yeah, my thoughts weren't were clear, as a few open questions regarding plugins were still lingering through my head. For example: You're right that when currently running the `installDist` task, core plugin JARs end up in the `lib` directory. This is in contrast to third-party plugins which by convention should be put into the `plugin` directory before running ORT. I was assuming that going forward it probably makes sense to move core plugins to the `plugin` directory, too, and was thinking about the case when something does wrong there, and not all plugin JARs are available. But that's probably something to be discussed once we've finalized the move of all plugins to the plugins Gradle project.